app/controllers/project/Versions.scala

```diff
@@ -198,8 +198,10 @@ class Versions @Inject()(stats: StatTracker,
             try {
               this.factory.processSubsequentPluginUpload(uploadData, user, request.project).fold(
                 err => Redirect(call).withError(err),
-                version =>
+                version => {
+                  version.underlying.authorId = user.id.getOrElse(-1)
```

Comment:
This assigns the uploader as the author of the version. This is the core line of the PR. The rest of it is adding `authorId` to the DB.